src/modules/trade/pure/TradeWidgetField/index.tsx

```diff
@@ -1,26 +1,31 @@
+import React from 'react'
+
 import { Trans } from '@lingui/macro'
 
 import QuestionHelper from 'legacy/components/QuestionHelper'
+import { renderTooltip } from 'legacy/components/Tooltip'
 
 import { TradeWidgetFieldBox, TradeWidgetFieldLabel, Content, ErrorText } from './styled'
 
 export type TradeWidgetFieldError = { type: 'error' | 'warning'; text: string | null } | null
 
 export interface TradeWidgetFieldProps {
-  label: string
+  label: React.ReactNode
   children?: JSX.Element
-  hint?: JSX.Element | string
+  tooltip?: React.ReactNode | ((params: any) => React.ReactNode)
   error?: TradeWidgetFieldError
   className?: string
 }
 
 export function TradeWidgetField(props: TradeWidgetFieldProps) {
-  const { className, children, label, hint, error } = props
+  const { className, children, label, tooltip, error } = props
+  const tooltipElement = renderTooltip(tooltip, props)
```

Comment:
Isn't this the only place where you'll ever have maybe a fn tooltip?
Everywhere else can just get the tooltip directly, no?